### PR TITLE
Use git branch --show-current because it works when there are no commits

### PR DESCRIPTION
**Pull request by rhoboat**
Needed for https://github.com/gruntwork-io/terraform-aws-architecture-catalog/pull/138

### CODE REVIEW

File: modules/git/git.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ func GetCurrentBranchName(t testing.TestingT) string {
 // GetCurrentBranchNameE retrieves the current branch name or
 // empty string in case of detached state.
 func GetCurrentBranchNameE(t testing.TestingT) (string, error) {
-	cmd := exec.Command("git", "rev-parse", "--abbrev-ref", "HEAD")
+	cmd := exec.Command("git", "branch", "--show-current")
```
Comment (yorinasub17) [inline modules/git/git.go:24]: This option was introduced in git 2.22, which unfortunately is newer than the default that is installable with Ubuntu 18.04. So I don't think we should switch to this command completely.

Instead, we should do a fallback approach. Either:

- Check the version of git installed and use this if newer than 2.22, or fallback to the other one
- Run `git branch --show-current` and if it fails, fall back to `git rev-parse --abbrev-ref HEAD`.

Comment (rhoboat) [reply modules/git/git.go:24]: Ah, good idea! Will change. Thanks for the review.

File: modules/git/git.go [hunk 2]
```diff
@@ -20,7 +20,36 @@ func GetCurrentBranchName(t testing.TestingT) string {
...
+			return "", nil
+		}
+
+		return name, nil
+	}
+
+	name := strings.TrimSpace(string(bytes))
+	if name == "HEAD" {
+		return "", nil
+	}
```
Comment (yorinasub17) [inline modules/git/git.go:45]: NIT: DRY with the logic above by extracting the git subprocess call into its own function.

**Review by yorinasub17 — COMMENTED** [review]
Had one minor nit for code style, but other than that LGTM!

**Comment by rhoboat** [comment]
I just force-pushed a rebase. Will take care of the nit now.

File: modules/git/git.go (hunk 2 above)
Comment (rhoboat) [reply modules/git/git.go:35]: Of course!

**Review by zackproser — APPROVED** [review]
LGTM

**Pull request merged**